src/Database/DriverInterface.php

```diff
@@ -302,4 +324,11 @@ public function newTableSchema(string $table, array $columns = []): TableSchemaI
      * @return int|null
      */
     public function getMaxAliasLength(): ?int;
+
+    /**
+     * Gets the the logger instance.
+     *
+     * @return \Psr\Log\LoggerInterface|null Logger instance if set.
+     */
+    public function getLogger(): ?LoggerInterface;
```

Comment:
I wish we could have `logQuery()` instead of `getLogger()`. The tricky part is do we also add a getter for `hasLogger()` so that the conditional `setLogger()` in `ConnectionHelper` can be functional. Another potential path forward could be to make `logQuery()` into:

```php
public function logQuery(string $message, array $context = []): bool
```

This would allow `ConnectionHelper` to do something like

```php
if ($this->connection->logQuery('--Starting test run ' . date('Y-m-d H:i:s'))) {
    $this->connection->setLogger(new QueryLogger());
}
```

This would also add a DX improvement of separating test runs with timestamps in the logs.